### PR TITLE
Feature/liquidation improvements

### PR DESCRIPTION
**Pull request by davidvuong**
- Adds exceed maxPd bypass
- Naming consistencies
- Plenty of bug fixes
- More tests (although still plenty of skipped tests for now)
- Endorsed keeper
- More configuration to support maxPd bypass and endorsed keeper

### CODE REVIEW

File: contracts/storage/Position.sol [hunk 1]
```diff
@@ -46,6 +46,15 @@ library Position {
...
     }
 
+    // --- Runtime structs --- //
+
+    struct Runtime_validateLiquidation {
+        address flagger;
+        uint128 oldPositionSize;
+        uint128 maxLiquidatableCapacity;
+        uint128 remainingCapacity;
+    }
```
Comment (bachstatter) [inline contracts/storage/Position.sol:56]: I like having the runtime struct close to where it's used. But I do remember seeing the Runtime structs getting defined in the interface in Perpsv3. Not sure if there any reasons for doing that, if not I prefer this

File: test/integration/modules/LiquidationModule.test.ts [hunk 2]
```diff
@@ -1,47 +1,520 @@
...
-    it('should flag a position with a health rating <= 1');
+    it('should flag a position with a health factor <= 1', async () => {
+      const { PerpMarketProxy } = systems();
 
-    it('should remove any pending orders when present');
+      const orderSide: 1 | -1 = genOneOf([1, -1]);
+      const { trader, market, marketId, collateral, collateralDepositAmount } = await depositMargin(bs, genTrader(bs));
+      const order = await genOrder(bs, market, collateral, collateralDepositAmount, {
+        desiredLeverage: 10,
+        desiredSide: orderSide,
```
Comment (bachstatter) [inline test/integration/modules/LiquidationModule.test.ts:23]: `getSide`

File: test/integration/modules/LiquidationModule.test.ts [hunk 3]
```diff
@@ -1,47 +1,520 @@
...
+      });
+
+      await commitAndSettle(bs, marketId, trader, order);
+
+      // Price falls/rises between 10% should results in a healthFactor of < 1.
+      //
+      // Whether it goes up or down depends on the side of the order.
+      const { answer: marketOraclePrice } = await market.aggregator().latestRoundData();
```
Comment (bachstatter) [inline test/integration/modules/LiquidationModule.test.ts:31]: We do this in most tests if we want to save some time in our tests I think we can grab `marketOraclePrice` from `order.oraclePrice`

File: test/integration/modules/LiquidationModule.test.ts [hunk 4]
```diff
@@ -1,47 +1,520 @@
...
+          .mul(orderSize === 1 ? 0.9 : 1.1)
+          .toBN()
+      );
+      await PerpMarketProxy.connect(keeper()).flagPosition(trader.accountId, marketId);
+
+      // Attempt the liquidate. This should complete successfully.
+      const tx = await PerpMarketProxy.connect(keeper()).liquidatePosition(trader.accountId, marketId);
+      const keeperAddress = await keeper().getAddress();
+
+      // NOTE: Missing the last two values (liqReward and keeperFee).
```
Comment (bachstatter) [inline test/integration/modules/LiquidationModule.test.ts:204]: This note wasn't very clear to me. Is it a TODO or are we excluding them on purpose?

**Review by bachstatter — APPROVED** [review]
NIce, left some comments but all minor

File: test/integration/modules/LiquidationModule.test.ts (hunk 2 above)
Comment (davidvuong) [reply test/integration/modules/LiquidationModule.test.ts:23]: Oops good catch

File: test/integration/modules/LiquidationModule.test.ts (hunk 4 above)
Comment (davidvuong) [reply test/integration/modules/LiquidationModule.test.ts:204]: Yep should be TODO. I'll update. Once I impl a better way to get liq/keeperFee then will update these.

**Pull request merged**